backend/src/resolvers/user_management.spec.js

```diff
@@ -309,3 +310,81 @@ describe('change password', () => {
     })
   })
 })
+
+describe('do not expose private RSA key', () => {
+  let headers
+  let client
+  const queryUserPrivateKey = gql`
+    query($queriedUserSlug: String) {
+      User(slug: $queriedUserSlug) {
+        id
+        privateKey
+      }
+    }`
+
+  const actionGenUserWithKeys = async () => {
+    // Generate user with "privateKey" via 'CreateUser' mutation instead of using the factories "factory.create('User', {...})", see above.
+    const variables = {
+      id: 'bcb2d923-f3af-479e-9f00-61b12e864667',
+      password: 'xYz',
+      slug: 'apfel-strudel',
+      name: 'Apfel Strudel',
+      email: 'apfel-strudel@test.org'
+    }
+    await client.request(gql`
+      mutation($id: ID, $password: String!, $slug: String, $name: String, $email: String) {
+        CreateUser(id: $id, password: $password, slug: $slug, name: $name, email: $email) {
+          id
+        }
+      }`, variables
+    )
+  }
+
+  describe('unauthenticated query of "privateKey"', () => {
+    it('throws "Not Authorised!"', async () => {
+      client = new GraphQLClient(host)
+
+      await actionGenUserWithKeys()
+      await expect(
+        client.request(queryUserPrivateKey, { queriedUserSlug: 'apfel-strudel' })
+      ).rejects.toThrow('Not Authorised')
+    })
+  })
+
+  // authenticate
+  beforeEach(async () => {
+    headers = await login({ email: 'test@example.org', password: '1234' })
+    client = new GraphQLClient(host, { headers })
+  })
+
+  describe('authenticated query of "privateKey"', () => {
+    it('throws "Not Authorised!"', async () => {
+      await actionGenUserWithKeys()
+      await expect(
+        client.request(queryUserPrivateKey, { queriedUserSlug: 'apfel-strudel' })
+      ).rejects.toThrow('Not Authorised')
+    })
+  })
+
+  describe('authenticated query of "publicKey". To asure generation of "privateKey" is garanteed', () => {
+    it('throws "Not Authorised!"', async () => {
+      const queryUserPuplicKey = gql`
+        query($queriedUserSlug: String) {
+          User(slug: $queriedUserSlug) {
+            id
+            publicKey
+          }
+        }`
+
+      await actionGenUserWithKeys()
+      await expect(
+        await client.request(queryUserPuplicKey, { queriedUserSlug: 'apfel-strudel' })
```

Comment:
@Tirokk it is important that `client` is unauthenticated here. For ActivityPub, unauthenticated 3rd-party servers like Mastodon and Friendica must be able to access an actor's public RSA key to check a the signature of e.g. a Post.

So please move this test case out of the `describe` block that logs in the `client`.

Reply:
Okay, I think I understand @roschaefer .
But do you think that any user data should be queried unauthenticated in the future? Like it is now …

Reply:
:man_shrugging: dunno. Everything except email or password is OK to be public for now. I think we need more user testing and we should also consult @datenbrei in these matters

Reply:
Okay. I noticed that as question to clarify.